repos/spack_repo/builtin/packages/py_geopmpy/package.py

```diff
@@ -13,30 +13,48 @@ class PyGeopmpy(PythonPackage):
 
     homepage = "https://geopm.github.io"
     git = "https://github.com/geopm/geopm.git"
-    url = "https://github.com/geopm/geopm/tarball/v3.1.0"
+    url = "https://github.com/geopm/geopm/tarball/v3.2.0"
 
     maintainers("bgeltz", "cmcantalupo")
     license("BSD-3-Clause")
     tags = ["e4s"]
 
     version("develop", branch="dev", get_full_repo=True)
+    version("3.2.0", sha256="b708233e1bfda66408c500f2ac0cbaf042140870bffdced12dd7cabbd18e0025")
     version("3.1.0", sha256="2d890cad906fd2008dc57f4e06537695d4a027e1dc1ed92feed4d81bb1a1449e")
 
+    for ver in ["3.1.0", "3.2.0", "develop"]:
+        depends_on(f"py-geopmdpy@{ver}", type="run", when=f"@{ver}")
+        depends_on(f"geopm-runtime@{ver}", type=("build", "run"), when=f"@{ver}")
+
     depends_on("python@3.6:3", type=("build", "run"))
-    depends_on("py-setuptools@53.0.0:", type="build")
+    depends_on("py-setuptools@53.0.0:", when="@3.1", type="build")
+    depends_on("py-setuptools@59.6.0:", when="@3.2:", type="build")
     depends_on("py-setuptools-scm@7.0.3:", when="@3.1:", type="build")
```

Comment:
This is a newer version of `py-setuptools-scm@6.4.2:` in the `requirements.txt` or `pyproject.toml` (https://github.com/geopm/geopm/blob/v3.2.0/geopmpy/pyproject.toml).

Reply:
I picked 7.0.3 because it's the first version that currently exists in the py_setuptools_scm/package.py that fulfills geopmpy requirements.txt (setuptools-scm>=6.4.2).  I could also add 6.4.2 to the package.py if that's preferred.  Otherwise, I can add a comment to the py_geopmpy/package.py to explain that detail.

Reply:
I generally suggest sticking to what the software requires. That handles the case of someone later adding older versions to the other packages. It also helps if a reviewer decides to look at the documented dependencies.

But feel free to add a comment explaining the reason for the difference.